lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,293 @@
+"""
+APIView endpoints for creating and bulk enrolling users
+"""
+import logging
+from uuid import uuid4
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email, validate_slug
+from django_countries import countries
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from enrollment.views import ApiKeyPermissionMixIn, EnrollmentCrossDomainSessionAuth, EnrollmentUserThrottle
+from lms.djangoapps.instructor.enrollment import enroll_email, get_email_params, get_user_email_language, unenroll_email
+from lms.djangoapps.instructor.views.tools import get_student_from_identifier
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from openedx.core.djangoapps.user_api.preferences.api import update_user_preferences
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from student.forms import PasswordResetFormNoActive
+from student.models import UserProfile
+from student.views import create_account_with_params
+from xmodule.modulestore.django import modulestore
+
+log = logging.getLogger(__name__)
+
+
+class CreateUserAccountWithoutPasswordView(APIView):
+    """
+    Create user account without password
+    """
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission,)
+
+    def post(self, request):
+        """
+        Create user account without password
+
+        Creates a user using mail, login and also name and surname.
+        Sets a random password and sends a user a message to change it.
+        """
+        data = request.data
+        data['honor_code'] = "True"
+        data['terms_of_service'] = "True"
+        email = request.data.get('email')
+        username = request.data.get('username')
+        # countries.by_name return country code or '' for else cases
+        country = countries.by_name(request.data.get('country', ''))
+        first_name = request.data.get('first_name', '')
+        last_name = request.data.get('last_name', '')
+        language = request.data.get('language')
+        if not username:
+            error_msg = "username is required parameter."
+        elif not email:
+            error_msg =  "email is required parameter."
+        elif not country:
+            error_msg = (
+                "Country is wrong or absent: {country}. For checking: Visit https://www.iso.org/obp . "
```

Comment:
I would rephrase s/Country is wrong or absent/Country is incorrect or missed/

Reply:
I changed the message.